gulpfile.babel.js

```diff
@@ -215,39 +217,57 @@ gulp.task('pagespeed', cb =>
 );
 
 // Copy over the scripts that are used in importScripts as part of the generate-service-worker task.
-gulp.task('copy-sw-scripts', () => {
-  return gulp.src(['node_modules/sw-toolbox/sw-toolbox.js', 'app/scripts/sw/runtime-caching.js'])
-    .pipe(gulp.dest('dist/scripts/sw'));
+gulp.task('third-party:dev', () => {
+  return gulp.src(['node_modules/workbox-sw/build/importScripts/workbox-sw.dev.*.js'])
+    .pipe($.rename('workbox-sw.dev.js'))
+    .pipe(gulp.dest('app/third_party/workbox-sw/'));
+});
+
+gulp.task('third-party:prod', () => {
+  return gulp.src(['node_modules/workbox-sw/build/importScripts/workbox-sw.prod.*.js'])
+    .pipe(gulp.dest('dist/third_party/workbox-sw/'));
 });
 
 // See http://www.html5rocks.com/en/tutorials/service-worker/introduction/ for
 // an in-depth explanation of what service workers are and why you should care.
 // Generate a service worker file that will provide offline functionality for
 // local resources. This should only be done for the 'dist' directory, to allow
 // live reload to work as expected when serving from the 'app' directory.
-gulp.task('generate-service-worker', ['copy-sw-scripts'], () => {
-  const rootDir = 'dist';
-  const filepath = path.join(rootDir, 'service-worker.js');
-
-  return swPrecache.write(filepath, {
-    // Used to avoid cache conflicts when serving on localhost.
-    cacheId: pkg.name || 'web-starter-kit',
-    // sw-toolbox.js needs to be listed first. It sets up methods used in runtime-caching.js.
-    importScripts: [
-      'scripts/sw/sw-toolbox.js',
-      'scripts/sw/runtime-caching.js'
-    ],
-    staticFileGlobs: [
-      // Add/remove glob patterns to match your directory setup.
-      `${rootDir}/images/**/*`,
-      `${rootDir}/scripts/**/*.js`,
-      `${rootDir}/styles/**/*.css`,
-      `${rootDir}/*.{html,json}`
+gulp.task('generate-service-worker', () => {
+  return workboxBuild.injectManifest({
+    swSrc: `app/service-worker.js`,
+    swDest: `dist/service-worker.js`,
+    globDirectory: `dist`,
+    // Add/remove glob patterns to match your directory setup.
+    globPatterns: [
+      `images/**/*`,
+      `scripts/**/*.js`,
+      `styles/**/*.css`,
+      `*.{html,json}`
     ],
     // Translates a static file path to the relative URL that it's served from.
     // This is '/' rather than path.sep because the paths returned from
     // glob always use '/'.
-    stripPrefix: rootDir + '/'
+    modifyUrlPrefix: {
+      'dist/': ''
```

Comment:
Is `dist/` actually showing up in the output URLs? It sounds like that's a bug if so, since it's also the `globDirectory` value.